app/src/pages/inside/projectSettingsPageContainer/content/defectTypes/modals/addEditDefectTypeModal/addEditDefectTypeModal.jsx

```diff
@@ -243,27 +245,35 @@ AddEditDefectTypeModal.propTypes = {
   }),
   initialize: PropTypes.func.isRequired,
   handleSubmit: PropTypes.func.isRequired,
-  dirty: PropTypes.bool.isRequired,
   change: PropTypes.func.isRequired,
+  currentValue: PropTypes.object,
+  initialValue: PropTypes.object,
 };
 AddEditDefectTypeModal.defaultProps = {
   data: {},
+  currentValue: {},
+  initialValue: {},
 };
 
 export const AddEditDefectTypeModalComponent = withModal('addEditDefectTypeModal')(
-  reduxForm({
-    form: 'DefectTypeForm',
-    validate: ({ longName, shortName }) => {
-      return {
-        longName: composeBoundValidators([
-          bindMessageToValidator(validate.required, 'shortRequiredFieldHint'),
-          bindMessageToValidator(validate.defectTypeLongName, 'defectLongNameHint'),
-        ])(longName),
-        shortName: composeBoundValidators([
-          bindMessageToValidator(validate.required, 'shortRequiredFieldHint'),
-          bindMessageToValidator(validate.defectTypeShortName, 'defectShortNameHint'),
-        ])(shortName),
-      };
-    },
-  })(AddEditDefectTypeModal),
+  connect((state) => ({
+    currentValue: getFormValues('DefectTypeForm')(state),
```

Comment:
My suggestion is to rewrite it using formValueSelector from `redux-form` and useSelector hooks.
And we can select only the value of the color field and compare it with defeType.color value.